### ISSUE TITLE
关于diffusion loss计算的问题

### ISSUE THREAD

**Issue by machengnan** [tsinghua-fib-lab/SPDiff#5]
请问下面加粗这里的计算有问题不？ 通常来说diffusion loss的计算是net输出与高斯做mseloss的，您这里net输出与x_0计算；原始diffusion学习的是每一步的噪声，这样改之后感觉网络学习的是t步到0步的噪声， 而不是t到t-1的噪声，没有很理解，谢谢回答；

def get_loss(self, x_0, curr=None, context=None, timestep=0.08, t=None, mask=None):

        # batch_size = x_0.shape[0]
        point_dim = x_0.shape[-1]
        x_0_hat = self.denoise_fn(x_0, curr=curr, context=context, timestep=timestep, t=t, mask=mask)

        if mask==None:
            **loss = F.mse_loss(x_0_hat.view(-1,point_dim), x_0.continuous().view(-1,point_dim), reduction='mean')**
        else:
            loss = mask_mse_func(x_0_hat, x_0, mask)       
        return loss

**Comment by machengnan** [comment]
原始diffusion学习的是每一步的噪声，这样改之后也是学习的每一步的噪声嘛？有理论解释不
